### ISSUE TITLE
Add the hits parameters

### ISSUE THREAD

**Issue by robsonpeixoto** [ExHammer/hammer#22]
Some solutions has the concept of query complexity/cost and I'd like to this value to rate limite the customer.

If change the api to add the `hits` parameter will be possible to use hammer to rate limit
```elixir
  @spec check_rate(id :: String.t(), scale_ms :: integer, limit :: integer, hits :: integer) ::
          {:allow, count :: integer}
          | {:deny, limit :: integer}
          | {:error, reason :: any}
  def check_rate(id, scale_ms, limit, hits \\ 1) do
    check_rate(:single, id, scale_ms, limit, hits)
  end
```

Does that make sense to you?

**Comment by ShaneKilkelly** [comment]
Ah, I see, this does make sense.

Just to to totally clear, this change would allow you to specify the number to be added to the bucket, rather than the constant `1` that it is now?

**Comment by ShaneKilkelly** [comment]
Cool, I'll take a swing at it in the next week or so. Or, PRs welcome. :+1:

**Comment by ShaneKilkelly** [comment]
Ok, I've found that adding a default parameter to `count_hit` causes a conflict between the two existing function-heads (`count_hit/3` and `count_hit/4`), so this feature will be added as a new function: `count_hit_inc`.

**Comment by ShaneKilkelly** [comment]
I have released version `5.0.0-rc1` of both hammer and hammer-backend-redis.

They include a new `check_rate_inc` function that can be used to supply a custom increment. Example:

```elixir
# Bulk file upload
user_id = get_user_id_somehow()
n = get_number_of_files()
case Hammer.check_rate_inc("upload_file_bulk:#{user_id}", 60_000, 10, n) do
  {:allow, _count} ->
    # upload all of the files
  {:deny, _limit} ->
    # deny the request
end
```

**Comment by ShaneKilkelly** [comment]
Cool, I'll cut a release for `5.0.0` soon. Thanks!

**Comment by ShaneKilkelly** [comment]
This is done, and released in `v5.0.0`, of both `hammer` and `hammer_backend_redis`.

There is a new tutorial section on [Custom Increments](https://hexdocs.pm/hammer/tutorial.html#custom-increments) which explains how to use this new feature.
